tracer.go

```diff
@@ -266,3 +264,17 @@ func (t Tags) Apply(o *StartSpanOptions) {
 		o.Tags[k] = v
 	}
 }
+
+// Tag may be passed as a StartSpanOption to add a tag to new spans.
+type Tag struct {
```

Comment:
I agree, although I think we should also rename the existing types xxxTag in tags.go to xxxTagName, since there are a bunch of `func (x xxxTag) Set(s Span, v xxx)` methods already, just for the sake of maintainability.